src/app/providers/profile/profile.ts

```diff
@@ -1974,6 +1980,64 @@ export class ProfileProvider {
     });
   }
 
+  createDefaultWalletsForSimpleFlow(isSkipAskEncrypt?: boolean) {
+    return new Promise((resolve, reject) => {
+      const coins = ['xpi', 'xec'];
+      const defaultOptsXpi = this.getDefaultWalletOpts(coins[0]);
+      defaultOptsXpi.name = `${defaultOptsXpi.name} - 1899`
+      defaultOptsXpi.isSlpToken = true;
+      this._createWallet(defaultOptsXpi).then(data => {
+        const key = data.key;
+        const firstWalletData = data;
+        const newOpts: any = {};
+        Object.assign(newOpts, this.getDefaultWalletOpts(coins[1]));
+        newOpts['key'] = key; // Add Key
+        newOpts['keyId'] = key.id;
+        newOpts['name'] += ' - 1899';
+        delete newOpts['mnemonic'];
+        newOpts['isSlpToken'] = true;
+        newOpts['singleAddress'] = true;
+        const create2ndWallets = [this._createWallet(newOpts)];
+        Promise.all(create2ndWallets)
+          .then(walletsData => {
+            walletsData.unshift(firstWalletData);
+            let walletClients = _.map(walletsData, 'walletClient');
+            const data = {
+              key: firstWalletData.key,
+              walletClients,
+              isSkipAskEncrypt: !!isSkipAskEncrypt
+            }
+            this.addAndBindWalletClients(data)
```

Comment:
I think we should flatten the promises instead of using promises with the nested approach.
Please try to use the promise pattern such as:
Promise.then(() => return x).then((x) => return y)...